src/wp-includes/class-wp-site-query.php

```diff
@@ -345,8 +345,10 @@ public function get_sites() {
 		// $args can include anything. Only use the args defined in the query_var_defaults to compute the key.
 		$_args = wp_array_slice_assoc( $this->query_vars, array_keys( $this->query_var_defaults ) );
 
-		// Ignore the $fields argument as the queried result will be the same regardless.
+		// Ignore the $fields, $update_site_cache, $update_site_meta_cache argument as the queried result will be the same regardless.
 		unset( $_args['fields'] );
+		unset( $_args['update_site_cache'] );
+		unset( $_args['update_site_meta_cache'] );
```

Comment:
With the argument count now up to 3, it may be time to use an `array_diff` of some type, between which keys should/not be included in the cache key generator.

* Maybe someday there will be a need to add more keys
* An explicit array of keys shows the intent of the code more than several similar lines of code back-to-back which may be more easily overlooked
* The additional verbosity allows more room for code comments
* Seeing as this pattern is emerging in multiple `WP_*_Query` classes, perhaps this is worthy of its own bonafide approach that can intentionally be shared between them for improved familiarity?

```php

		// Arguments that should not alter the cache key
		$no_cache_args = array(
			'fields',
			'update_site_cache',
			'update_site_meta_cache'
		);

		// Strip the arguments
		$_args = array_diff( $_args, $no_cache_args );
```

----

In addition, `unset()` accepts multiple arguments, so perhaps this could even just be...

```php
		unset(
			$_args['fields'],
			$_args['update_site_cache']
			$_args['update_site_meta_cache']
		);
```

----

I prefer the first approach, fwiw.

Reply:
I personally find the unset much easier to read and understand. 

I agree, I think that all the query classes should be more structured and have methods like get_cache_key. But WordPress core has been against refactoring for the sake of refactoring. I have a ticket on this matter [#37463](https://core.trac.wordpress.org/ticket/37463) if anyone is interested.